core/dbt/contracts/relation.py

```diff
@@ -120,3 +109,22 @@ def get_lowered_part(self, key: ComponentName) -> Optional[str]:
         if part is not None:
             part = part.lower()
         return part
+
+    def get_part(self, key: ComponentName) -> Optional[str]:
```

Comment:
this is definitely more code that the previous version, but it is _so much_ clearer! Big fan of this kind of change